File: app/src/test/java/com/example/moviemvp/moviedetail/MovieDetailsPresenterTest.java
package com.example.moviemvp.moviedetail;

import com.example.moviemvp.model.Movie;

import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;

public class MovieDetailsPresenterTest {
    private MovieDetailsPresenter presenter;
    @Mock
    private MovieDetailsModel model;
    @Mock
    private MovieDetailsActivity view;
    private int movieId;
    private Throwable t;
    private Movie movie;
    private MovieDetailsContract.Model.OnFinishedListener a;

    @Before
    public void setUp() throws Exception {
        MockitoAnnotations.initMocks(this);
        presenter = new MovieDetailsPresenter(view, model);
    }

    @Test
    public void onDestroy() {
        presenter.onDestroy();
    }

    @Test
    public void requestMovieData() {
        presenter.requestMovieData(movieId);
        model.getMovieDetails(a, movieId);
    }

    @Test
    public void onFinished() {
        presenter.onFinished(movie);
        view.setDataToViews(movie);
    }

    @Test
    public void onFailure() {
        presenter.onFailure(t);
        view.onResponseFailure(t);
    }
}

File: app/src/test/java/com/example/moviemvp/movie_detail/MovieDetailsPresenterTest.kt
package com.example.moviemvp.movie_detail

import com.example.moviemvp.model.Movie
import org.junit.Assert
import org.junit.Before
import org.junit.Test
import org.mockito.Mock
import org.mockito.MockitoAnnotations

class MovieDetailsPresenterTest {

    private lateinit var presenter: MovieDetailsPresenter
    @Mock
    private lateinit var view: MovieDetailsContract.View
    private var t: Throwable = Throwable()
    private var movie=Movie(1,"","",1F,"","",""
    ,"","","")
    @Before
    fun setUp() {

        MockitoAnnotations.initMocks(this)
        presenter = MovieDetailsPresenter(view)
    }

    @Test
    fun onDestroy() {

        presenter.onDestroy()
    }

    @Test
    fun requestMovieData() {

        presenter.requestMovieData(1)
    }

    @Test
    fun onFinished() {

        presenter.onFinished(movie)
    }

    @Test
    fun onFailure() {

        presenter.onFailure(t)
        Assert.assertEquals(view.onResponseFailure(t), Unit)
    }
}